2016/proposals/stolliker-ryan.md

```diff
@@ -17,22 +13,39 @@ All server management will be written in Python. The server itself will be imple
 ####Interactions
 
 * `/first/`
-  * The first installation script will send its data here in JSON for forms format. The server will send back a 200 response.
+  * The first installation script will send platform, system, release, implementation, version information here in JSON or forms format. The server will send back a 200 response.
 * `/second/`
-  * The second installation script will send its data here in JSON for forms format. The server will send back a 200 response.
+  * The second installation script will send platform, system, release, implementation, version and dependency data here in JSON or forms format. The server will send back a 200 response.
 * `/data/`
-  * A request to this URL will recieve JSON from the server which contains information from the database for analysis in some future application.
+  * A request to this URL will recieve JSON from the server which contains all information from the database for analysis in some future application.
+
+###Database
+
+SQLite will be used as the database to store the results of the installation scripts. SQLAlchemy will be used to manage the database and all Python interaction with the database. If database migration is needed in the future, Alembic will be used.
+
+####Proposed Schema
+
+* First(FID, platform, system, release, implementation, version)
+  * FID is a unique integer automatically assigned by the database when a tuple is inserted, and is the primary key of the relation.
+  * platform is the system information provided by calling `platform.platform`
+  * system is the operating system provided by `platform.system`
+  * release is the version of the operating system provided by `platform.release`
+  * implementation is the Ptyhon distribution (Cpython, Pypy, etc.) provided by `platform.python_implementation`
+  * version is the Python version provided by `platform.python_version`
+* Second(SID, platform, system, release, implementation, version, *all tests*)
+  * SID, platform, system, release, implementation, and version are hold almost the same information as in First, however, if someone saw that they were failing the first test script and got a newer version of the interpreter, then the results may be different and therefore should be collected again in the second test script.
+  * *all tests* refers to the pass/fail status of each dependency. Each library or program is a boolean showing wether or not it is installed correctly. If one is null, that means that the dependency was not tested for (since individual workshops can change which ones they need)
```

Comment:
Regarding "Proposed Schema" section: I'd like you to write exactly how do you envision your database tables look like. For example: it seems you want to store tests horizontally, ie. each test (for example for installed `make` tool) is in its own column (I might not be reading your statements correctly, though).

Such approach would not scale well; it's better to have each test as a record in one table and tests from the same computer should be grouped together (possibly via a relationship [your `FID`?]).

Do you have experience in database systems? If not, Software-Carpentry has a lesson on SQLite http://swcarpentry.github.io/sql-novice-survey/.